crates/node/builder/src/builder/mod.rs

```diff
@@ -154,28 +154,30 @@ pub type RethFullAdapter<DB, Types> = FullNodeTypesAdapter<
 /// configured by the builder itself during launch. This might change in the future.
 ///
 /// [builder]: https://doc.rust-lang.org/1.0.0/style/ownership/builders.html
-pub struct NodeBuilder<DB> {
+pub struct NodeBuilder<DB, ChainSpec> {
     /// All settings for how the node should be configured.
-    config: NodeConfig,
+    config: NodeConfig<ChainSpec>,
```

Comment:
there's perhaps an argument to make the entire config generic,so the builder can be used with custom config, but we should investigate this separately